damage.py

```diff
@@ -21,6 +21,27 @@ def pre_mitigation_damage(
     return (base_damage + bonus_damage) * damage_modifier_percent_mult_factor + damage_modifier_flat
 
 
+def pre_mitigation_spell_damage(
+    base_spell_damage: float,
+    ratio: float,
+    base_damage: float,
+    bonus_damage: float,
+    damage_modifier_flat: float = 0,
+    damage_modifier_percent_mult_factor: float = 1,
+):
+    """
+    Calculates the pre-mitigation damage of a spell or an autoattack
+    All values regarding damage modifiers should include the buffs/debuffs coming from spells, summoner spells, or items
+    from both the attacker AND the defender
+    """
+
+    return (
+        base_spell_damage
+        + ratio * (base_damage + bonus_damage) * damage_modifier_percent_mult_factor
+        + damage_modifier_flat
+    )
+
+
 def avg_pre_mitigation_physical_damage(
```

Comment:
maybe crit should appear in the name to make it clear that it is an average based on crit chance and crit damage.